src/compressor/Compressor.h

```diff
@@ -16,7 +16,10 @@
 #define CEPH_COMPRESSOR_H
 
 #include "include/int_types.h"
-#include "include/Context.h"
+#include "include/memory.h"
```

Comment:
should use `<memory>` here instead. it's portable. we added our own `memory.h` back in the pre-C++11 era, when the smart pointers were still in TR1.